pkg/dataobj/internal/encoding/pools.go

```diff
@@ -0,0 +1,20 @@
+package encoding
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+var bytesBufferPool = sync.Pool{
+	New: func() any {
+		return new(bytes.Buffer)
+	},
+}
+
+var protoBufferPool = sync.Pool{
+	New: func() any {
+		return new(proto.Buffer)
```

Comment:
@cyriltovena I initially set `proto.Buffer.SetDeterministic` here to have deterministic encoding of protobufs but I think there's a bug in gogo protobuf that prevents it from working.

Either way, I think our encoding is already deterministic as long as we never include map types in our protobuf. I'll have some tests for that once we have the final pieces that tie everything together.